sdk/src/azure/platform/az_mosquitto5.c

```diff
@@ -315,50 +315,50 @@ az_mqtt5_outbound_connect(az_mqtt5* mqtt5, az_mqtt5_connect_data* connect_data)
 
   if (az_span_ptr(connect_data->username) != NULL)
   {
-    _az_RETURN_IF_FAILED(_az_result_from_mosq(mosquitto_username_pw_set(
+    _az_RETURN_IF_FAILED(_az_result_from_mosq5(mosquitto_username_pw_set(
         me->_internal.mosquitto_handle,
         (const char*)az_span_ptr(connect_data->username),
         (const char*)az_span_ptr(connect_data->password))));
   }
 
-  _az_RETURN_IF_FAILED(_az_result_from_mosq(mosquitto_connect_async(
+  _az_RETURN_IF_FAILED(_az_result_from_mosq5(mosquitto_connect_async(
       (struct mosquitto*)me->_internal.mosquitto_handle,
       (char*)az_span_ptr(connect_data->host),
       connect_data->port,
       AZ_MQTT5_DEFAULT_MQTT_CONNECT_KEEPALIVE_SECONDS)));
 
-  _az_RETURN_IF_FAILED(_az_result_from_mosq(mosquitto_loop_start(me->_internal.mosquitto_handle)));
+  _az_RETURN_IF_FAILED(_az_result_from_mosq5(mosquitto_loop_start(me->_internal.mosquitto_handle)));
 
   return ret;
 }
 
 AZ_NODISCARD az_result az_mqtt5_outbound_sub(az_mqtt5* mqtt5, az_mqtt5_sub_data* sub_data)
 {
-  return _az_result_from_mosq(mosquitto_subscribe_v5(
+  return _az_result_from_mosq5(mosquitto_subscribe_v5(
       mqtt5->_internal.mosquitto_handle,
       &sub_data->out_id,
       (char*)az_span_ptr(sub_data->topic_filter),
       sub_data->qos,
       0,
-      sub_data->properties ? sub_data->properties->properties : NULL));
+      (sub_data->properties == NULL) ? NULL : sub_data->properties->properties));
 }
 
 AZ_NODISCARD az_result az_mqtt5_outbound_pub(az_mqtt5* mqtt5, az_mqtt5_pub_data* pub_data)
 {
-  return _az_result_from_mosq(mosquitto_publish_v5(
+  return _az_result_from_mosq5(mosquitto_publish_v5(
       mqtt5->_internal.mosquitto_handle,
       &pub_data->out_id,
       (char*)az_span_ptr(pub_data->topic), // Assumes properly formed NULL terminated string.
       az_span_size(pub_data->payload),
       az_span_ptr(pub_data->payload),
       pub_data->qos,
       false,
-      pub_data->properties ? pub_data->properties->properties : NULL));
+      (pub_data->properties == NULL) ? NULL : pub_data->properties->properties));
```

Comment:
This is hard to parse: as noted above, I would rename the Mosquitto-specific properties making this `pub_data->properties->mosq_property_handler` or similar.